### ISSUE TITLE
Actions pinned to tag SHA failing

### ISSUE THREAD

**Issue by beornf** [mheap/pin-github-action#21]
As discussed in #13 actions pinned to tag SHA have started failing on GitHub:

```
uses: actions/checkout@f1d3225b5376a0791fdee5a0e8eac5289355e43a # pin@v2
```

This is fixed by changing to the commit SHA:

```
uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # pin@v2
```

You can follow the discussion on the GitHub community forum https://github.community/t/github-workflows-started-to-fails/185188. The GitHub documentation only mentions commit SHA and the behavior for tag SHA seems uncertain with a possible breaking change.

**Comment by lukasbestle** [comment]
We (Kirby team) were able to verify that there was indeed a behavior change in GitHub's action runner yesterday. Until Monday the tag refs worked just fine (even though unsupported), but since yesterday they no longer work.

Switching to the commit SHAs manually did indeed fix the issue for us.

**Comment by mheap** [comment]
I've just released v1.3.0 that detects if the provided ref is a tag and resolves the commit for that tag if so. Thanks for the heads up!

**Comment by afbora** [comment]
@mheap Thanks for fixing. GH support team replied the thread on forum:

> This was broken by an internal change we made, sorry about that. I’ve disabled that change, so referencing a tag by the SHA object ID should work for now.

> As referenced in our documentation, the intended use is using a commit SHA rather than other objects.

https://github.community/t/github-workflows-started-to-fails/185188/10?u=afbora
